### ISSUE TITLE
Support for AWS SDK for JavaScript v3

### ISSUE THREAD

**Issue by anvar** [aws/aws-xray-sdk-node#294]
Is there a branch parked somewhere with support for the new [JavaScript AWS SDK](https://github.com/aws/aws-sdk-js-v3)? The new SDK uses middleware to intercept and transform requests/responses, which is a bit different to how the xray-sdk hooks into the 2.x version. If not, happy to get started and send a PR, just thought I would check beforehand.

Thanks!

**Comment by willarmiros** [comment]
Hi @anvar, 
We are not yet developing support for AWS SDK for JS v3. We'd be open to a pull request if you're interested. The only caveats are that we'd like to keep dependencies to an absolute minimum, although that shouldn't be an issue with v3's modular design. We'll also need to discuss with the AWS SDK team about the stability of the middleware APIs before we can mainline our own middleware, since they are still in beta.

Otherwise, contribute away!

**Comment by anvar** [comment]
Great, thank you, I will have a look!

**Comment by trivikr** [comment]
Update: The [general availability of the AWS SDK for JavaScript, version 3 (v3)](https://aws.amazon.com/blogs/developer/modular-aws-sdk-for-javascript-is-now-generally-available/) was announced on December 15th, 2020.

**Comment by awssandra** [comment]
Reviewing pull request in flight.

**Comment by nickitza** [comment]
Hey there, saw this was merged to master, but am getting the error "AWSXRay.captureAWSv3Client is not a function" after implementing the S3 client module with AWSXRay in a Lambda. Does version 3.2.0 contain these changes? Thanks!

**Comment by bfaulk96** [comment]
Hey all, I noticed https://github.com/aws/aws-xray-sdk-node/pull/386 got merged, but it seems I still get the `TypeError: service.customizeRequests is not a function` error when trying to capture the Dynamo Document Client from the sdk v3... Any updates?

**Comment by willarmiros** [comment]
@bfaulk96 see #395

**Comment by willarmiros** [comment]
Released in 3.3.0!

**Comment by willarmiros** [comment]
Reopening because we had to temporarily rollback AWS SDK V3 support in patch release v3.3.1  to address #411. We are actively working on another patch that will add back AWS SDK v3 support in 3.3.2.

**Comment by bfaulk96** [comment]
Any updates on this being re-implemented?

**Comment by awssandra** [comment]
We are actively addressing this to re-release. Stay tuned!

**Comment by jwoehrle** [comment]
I've tested some of my code with 3.3.3... works now as expected :tada: 

Thank you!

**Comment by willarmiros** [comment]
As @jwoehrle, AWS SDK v3 support is here to stay as of v3.3.3. Closing this out.
